plugins/mac-avcapture/OBSAVCapture.m

```diff
@@ -700,21 +628,67 @@ + (NSString *)stringFromSettings:(void *)settings withSetting:(NSString *)settin
     return result;
 }
 
++ (const char *)effectsWarningForDevice:(AVCaptureDevice *)device
+{
+    int effectsCount = 0;
+    const char *effectWarning = "";
+    if (@available(macOS 12.0, *)) {
+        if (device.portraitEffectActive) {
+            effectWarning = obs_module_text("PortraitEffectWarning");
+            effectsCount++;
+        }
+    }
+    if (@available(macOS 12.3, *)) {
+        if (device.centerStageActive) {
+            effectWarning = obs_module_text("CenterStageWarning");
+            effectsCount++;
+        }
+    }
+    if (@available(macOS 13.0, *)) {
+        if (device.studioLightActive) {
+            effectWarning = obs_module_text("StudioLightWarning");
+            effectsCount++;
+        }
+    }
+    /* This property is currently unavailable due to an SDK issue: FB13948132
+    if (@available(macOS 14.0, *)) {
+        if (device.reactionEffectGesturesEnabled) {
+            effectWarning = obs_module_text("ReactionsWarning");
+            effectsCount++;
+        }
+    }
+    */
+    /* Not available until we are building on Xcode 16. Is there a way to check SDK availability?
+    if (@available(macOS 15.0, *)) {
+        if (device.backgroundReplacementActive) {
+            effectWarning = obs_module_text("BackgroundReplacementWarning");
+            effectsCount++;
+        }
+    }
+     */
```

Comment:
> Also curious if there is a good way to check compile-time SDK availability, since the background replacement code that I've commented out here will only compile with the macOS 15 SDK.

@jcm93 like this:
```c
#if __MAC_OS_X_VERSION_MAX_ALLOWED >= 150000
...
#endif
```

At a later point we could raise the minimum version but since Xcode 16 is still in Beta 1 I feel like that's to early, the current minimum is this:
https://github.com/obsproject/obs-studio/blob/5f98d34e2c4d81a43eafd3d87ae65e67a3d04e06/cmake/macos/compilerconfig.cmake#L26-L27


On GHA we're on `macos-14` with macOS 14.5 and Xcode 15.2 (see the [docs](https://github.com/actions/runner-images/blob/main/images/macos/macos-14-arm64-Readme.md), default is 15.0.1 but we [change](https://github.com/obsproject/obs-studio/blob/5f98d34e2c4d81a43eafd3d87ae65e67a3d04e06/.github/workflows/build-project.yaml#L84) that), so it wouldn't be compiled anyways. GHA does provide Xcode 16 though if we're happy to compile on a Beta versions on CI, historically we've done that a couple of times.

Reply:
Thanks; added this check to the background replacement block.